### PR TITLE
Add support hashdb-3.1.0

### PR DESCRIPTION
**Pull request by linxon**
add support [hashdb-3.1.0](https://github.com/NPS-DEEP/hashdb/releases/tag/v3.1.0) and other minor changes

### CODE REVIEW

**Comment by simsong** [comment]
I have been told that there are still users running the 3.0 API and that that are incompatible with the 3.1 API. So I will need to make sure that this version runs with both.


----
Sent from my phone.

On Mar 20, 2019, at 6:13 AM, Yury Martynov <notifications@github.com> wrote:

add support hashdb-3.1.0 and other minor changes

You can view, comment on, or merge this pull request online at:

  https://github.com/simsong/bulk_extractor/pull/109

Commit Summary

fix: https://github.com/simsong/bulk_extractor/issues/89
File Changes

M src/scan_exiv2.cpp (8)
M src/scan_hashdb.cpp (39)
Patch Links:

https://github.com/simsong/bulk_extractor/pull/109.patch
https://github.com/simsong/bulk_extractor/pull/109.diff

**Comment by linxon** [comment]
Unfortunately it does not work to compile with version 3.0.0

Selection_302

**Comment by simsong** [comment]
Okay. Then we will need to have both versions present and have it automatically check with #ifdef statements. Can you do this? It should not be hard.


----
Sent from my phone.

On Mar 20, 2019, at 7:08 AM, Yury Martynov <notifications@github.com> wrote:

Unfortunately it does not work to compile with version 3.0.0

**Comment by simsong** [comment]
I should add that I apologize for the incompatibility; I was not aware that hashdb 3.1 did not include covers for the older API.



On Mar 20, 2019, at 7:12 AM, Simson Garfinkel <simsong@gmail.com> wrote:

Okay. Then we will need to have both versions present and have it automatically check with #ifdef statements. Can you do this? It should not be hard.


----
Sent from my phone.

On Mar 20, 2019, at 7:08 AM, Yury Martynov <notifications@github.com> wrote:

Unfortunately it does not work to compile with version 3.0.0

File: configure.ac [hunk 1]
```diff
@@ -403,6 +403,9 @@ if test x"$hashdb" == x"yes" ; then
 fi
 if test x"$hashdb" == xyes ; then
   AC_DEFINE(HAVE_HASHDB,1,[define 1 to build the hashdb scanner])
+  if [[ $(hashdb -v | cut -d' ' -f2) == "3.1.0" ]]; then
```
Comment (linxon) [inline configure.ac]: unfortunately i have not found another variant

**Pull request merged**